contrib/lkt/language/parser.py

```diff
@@ -3241,6 +3251,42 @@ class IfExpr(Expr):
     alternatives = Field(type=T.ElsifBranch.list)
     else_expr = Field(type=T.Expr)
 
+    @langkit_property(return_type=T.Expr.entity.array)
+    def result_exprs():
+        """
+        All expressions that compute a result.
+        """
+        return (
+            Entity.then_expr.singleton
+            .concat(Entity.alternatives.map(lambda alt: alt.then_expr))
+            .concat(Entity.else_expr.singleton)
+        )
+
+    @langkit_property()
+    def expr_context_free_type():
+        # Return the context type of the first alternative that has a non-null
+        # one.
+        #
+        # TODO: we should actually get the most-specific common ancestor of all
+        # context free types (for nodes).
+        return (
+            Entity.result_exprs
+            .map(lambda expr: expr.expr_context_free_type)
+            .find(lambda t: Not(t.is_null))
+        )
+
+    @langkit_property(return_type=T.TypeDecl.entity)
+    def if_expected_type():
+        """
+        Helper to handle ``if`` expressions in ``Expr.expected_type``.
+        """
+        # Use the if expression context free type if available, otherwise
+        # fallback on its expected type.
+        return Entity.expr_context_free_type.then(
```

Comment:
you can use `._or` here

Reply:
I didn’t know about it, thanks! Done, now.